### DISCUSSION TITLE
Automaterialization policy not triggering when upstream is non-partitioned and freshness policy is also not working

### DISCUSSION THREAD

**Discussion by sandeepmullangi2** [dagster-io/dagster#19710] [Q&A]
Hi Team,
I am working on POC on dagster and I am creating downstream table from 1 partitioned table and 1 non-partitioned table. I want my downstream table to trigger if all the upstream partitioned tables have updated data (in this case 1 table). I came up with automaterialization policy. I think non-partitioned table is causing problem here and not allowing downstream to be triggered.

```
auto_materialize_policy = AutoMaterializePolicy.eager(
                max_materializations_per_minute=100

            ).with_rules(
                AutoMaterializeRule.skip_on_not_all_parents_updated(require_update_for_all_upstream_partitions=True)
            ).without_rules(
                AutoMaterializeRule.skip_on_parent_missing(),
                AutoMaterializeRule.skip_on_parent_outdated()
            )
```

I have freshness policy as backup so that my downstream triggers atleast after time specified and it is also not working. i have following freshness policy

`freshness_policy = FreshnessPolicy(maximum_lag_minutes=1)
`
My downstream model says overdue I think precedence is given to materialization policy (1 of the skip condition is met) and then freshness is not triggered because of skip condition is met. In this case then what is the use of freshness policy if it is following materialization strictly? 
How do i say i dont want to depend on upstream non-partitioned asset for automaterializing downstream.

![image (6)](https://github.com/dagster-io/dagster/assets/70845449/bb86cf08-f005-4aa5-87c9-9a50e5d5080e)

**Comment by OwenKephart** [comment]
Hi @sandeepmullangi2 -- to clarify your setup a bit more, what is the precise scenario in which you want the downstream partitioned asset to be triggered?

Is it essentially just that every single partition of your upstream asset has been materialized more recently than the downstream (with the materialization of the unpartitioned asset being irrelevant to if the downstream gets kicked off or not)? 

What are the PartitionsDefinitions for the upstream / downstream asset, and do you have a non-default PartitionMapping between them?

**Reply by sandeepmullangi2** [reply]
@OwenKephart I want to trigger my downstream when all partitioned upstream tables have latest partitions than my downstream table, i want to trigger my downstream table. I do not want to consider non-partitioned table for my scheduling. Also if this is not right what is the acceptable/prevailing solution when i have multiple upstream partitioned and non partitioned tables which creates partitioned table downstream?
We do not have any non-default partition mapping atleast for the scenario we have issue with.

**Reply by sandeepmullangi2** [reply]
@OwenKephart We are waiting for your response. Any help is appreciated.

**Reply by OwenKephart** [reply]
Hi @sandeepmullangi2, apologies for the delay. Unfortunately, it's not currently possible to target your policy at a specific upstream asset, although this is functionality that has been requested in the past and is on our radar.

I think the best current workaround to achieve your desired behavior, considering these assets are near the root of your graph would be the following:

```python
from dagster import StaticPartitionsDefinition, asset
from dagster._core.definitions.auto_materialize_policy import AutoMaterializePolicy


@asset
def dummy_upstream() -> None:
    ...


@asset(
    partitions_def=StaticPartitionsDefinition(["a", "b", "c"]),
    deps=[dummy_upstream],
    auto_materialize_policy=AutoMaterializePolicy.eager(max_materializations_per_minute=None),
)
def partitioned_upstream():
    ...


@asset
def unpartitioned_upstream():
    ...


@asset(
    deps=[partitioned_upstream, unpartitioned_upstream],
    auto_materialize_policy=AutoMaterializePolicy.eager(),
)
def downstream():
    ...

```

In this case, I added a dummy unpartitioned asset  as an upstream to the partitioned parent. This asset can updated before each round of materializations for the static-partitioned parent. This way, the built-in "skip_on_parent_outdated" rule will prevent the downstream asset from materializing until all partitions of its parent have incorporated the latest materialization of that root unpartitioned asset, resulting in your desired behavior, albeit in a roundabout way.
